### ISSUE TITLE
Appending content to a view block that hasn't started is odd

### ISSUE THREAD

**Issue by kaliel86** [cakephp/cakephp#14253]
This is a (multiple allowed):

* [] bug
* [] enhancement
* [x] feature-discussion (RFC)

* CakePHP Version: 4.0.2
* Platform and Target: Arch/mariadb
### What you did
I want to fetch a view block at the bottom of my layout (templates/layout/default.php) : 
```
<html lang="klingon">
<head></head>
<body>

<main>
    <?= $this->Flash->render() ?>
    <?= $this->fetch('content') ?>
</main>

<footer></footer>

<?= $this->fetch('microdata'); ?>
</body>
</html>
```

Then in my action templates, i'd like to append data to this block (templates/products/index.php) with :
```
$this->assign('microdata', $jsonString);
```

### What happened
Nothing gets printed in te rendered html.
It works only if i manually end the block with ```$this->end();``` because the concat() method doesn't end the block itself.
In other words we could not append something to a block if it hasn't been started **and** ended first.

### What you expected to happen
I expected the block that i consider as an area to be filled with my string when i append something to it.

**Comment by garas** [comment]
Make sure `$jsonString` is not `null`, otherwise `$this->append('microdata', $jsonString = null);` starts capturing until finished with `$this->end();`.

Typecast value with `(string)` if value can be `null`.

**Comment by kaliel86** [comment]
i did, i even trigger this with a simple ```<p>yolo</p>```

**Comment by othercorey** [comment]
@kalessil You need to give us more specific information. This situation is in the unit tests.

**Comment by kalessil** [comment]
@kaliel86, looks like the previous comment is for you =)
